core/includes/install.core.inc

```diff
@@ -1454,6 +1450,367 @@ function install_select_language_form($form, &$form_state, $files) {
+        'description' => st('The installer requires write permissions to the <code>%translations_directory<code> directory during the installation process. The <a href="!handbook_url">File permissions and ownership</a> documentation section offers help on this and other topics.', array(
+          '%translations_directory' => $translations_directory,
+          '!handbook_url' => 'https://docs.backdropcms.org/documentation/file-permissions-and-ownership',
+        )),
+      );
+    }
+    else {
+      $requirements['translations directory writable'] = array(
+        'title'       => st('Translations directory'),
+        'value'       => st('The translations directory is writable.'),
+      );
+    }
+  }
+
+  // If the translations server can not be contacted, throw an error.
+  if (!$online) {
+    $requirements['online'] = array(
+      'title'       => st('Internet'),
+      'value'       => st('The translation server is offline or cannot be reached.'),
+      'severity'    => REQUIREMENT_ERROR,
+      'description' => st('The installer was unable to contact the translation server to download a translation file. Check your internet connection and verify that your website can reach the translation server at <a href="!server_url">@server_url</a>.', array(
+        '!server_url' => $server_url,
+        '@server_url' => $server_url,
+      )),
+    );
+  }
+  else {
+    $requirements['online'] = array(
+      'title'       => st('Internet'),
+      'value'       => st('The translation server is online.'),
+    );
+    // If translation file is not found at the translation server, throw an
+    // error.
+    if (!$translation_available) {
+      $requirements['translation available'] = array(
+        'title'       => st('Translation'),
+        'value'       => st('The %language translation is not available.', ['%language' => $language]),
+        'severity'    => REQUIREMENT_ERROR,
+        'description' => st('The %language translation file is not available at the translation server. <a href="!url">Choose a different language</a> or select English and translate your website later.', array(
+          '%language' => $language,
+          '!url' => $_SERVER['SCRIPT_NAME'],
+        )),
+      );
+    }
+    else {
+      $requirements['translation available'] = array(
+        'title'       => st('Translation'),
+        'value'       => st('The %language translation is available.', ['%language' => $language]),
+      );
+    }
+  }
+
+  if ($translations_directory_exists && $readable && $writable && $translation_available) {
+    $translation_downloaded = install_retrieve_file($translation_url, $translations_directory);
+
+    if (!$translation_downloaded) {
+      $requirements['translation downloaded'] = array(
+        'title'       => st('Translation'),
+        'value'       => st('The %language translation could not be downloaded.', ['%language' => $language]),
+        'severity'    => REQUIREMENT_ERROR,
```

Comment:
A warning should be fine here. OK, we couldn't grab the translation file, but... should we really stop the install because of that?
What do you think?

Reply:
Agreed edge case but agreed leave it in.